samples/drivers/lora/receive/README.rst

```diff
@@ -0,0 +1,55 @@
+.. zephyr:code-sample:: lora-receive
+   :name: LoRa receive
+   :relevant-api: lora_api
+
+   Receive a preset number of packets in both synchronous and asynchronous mode
+   using the LoRa radio.
+
+Overview
+********
+
+This sample demonstrates how to use the LoRa radio driver to receive packets 
+both synchronously and asynchronously.
+
+In order to successfully receive messages, build and flash the accompanying
+LoRa send sample :zephyr:code-sample:`lora-send` on another board within range.
+
+As this sample receives a finite number of packets and then sleeps infinitely,
```

Comment:
It feels like the sample should just asynchronously listen forever? Thoughts @JordanYates?

Reply:
I can change that if you wish

Reply:
Let's see what @JordanYates has to say first

Reply:
> Let's see what @JordanYates has to say first

I have just realized that the limitation imposed here may be to prevent someone accidentally leaving a transmitter on and not complying with ISM band rules.

Reply:
Oh wait, forget that - it's the wrong way around 😄

Reply:
> It feels like the sample should just asynchronously listen forever? Thoughts @JordanYates?

When I added the async listen support, I updated the sample to stop receiving after N packets so that there would be an example in code of how to do that. If someone wants to run the receiver forever, it is trivial to comment out the three lines. Less so to dig into the code to find out how to stop it.

Reply:
> I have just realized that the limitation imposed here may be to prevent someone accidentally leaving a transmitter on and not complying with ISM band rules.

Whether the receiver is listening or not has no effect on the transmitter violating ISM band limits :)

Reply:
> > I have just realized that the limitation imposed here may be to prevent someone accidentally leaving a transmitter on and not complying with ISM band rules.
> 
> Whether the receiver is listening or not has no effect on the transmitter violating ISM band limits :)

Yep - I had it the wrong way around 😿